lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/VideoUnitResource.java

```diff
@@ -0,0 +1,169 @@
+     */
+    @GetMapping("lectures/{lectureId}/video-units/{videoUnitId}")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<VideoUnit> getVideoUnit(@PathVariable Long videoUnitId, @PathVariable Long lectureId) {
+        log.debug("REST request to get VideoUnit : {}", videoUnitId);
+        Optional<VideoUnit> optionalVideoUnit = videoUnitRepository.findById(videoUnitId);
+        if (optionalVideoUnit.isEmpty()) {
+            throw new EntityNotFoundException("There isn't such video unit");
+
+        }
+        VideoUnit videoUnit = optionalVideoUnit.get();
+        if (videoUnit.getLecture() == null || videoUnit.getLecture().getCourse() == null) {
+            throw new ConflictException();
+        }
+        if (!videoUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException();
+        }
+        if (!authorizationCheckService.isAtLeastEditorInCourse(videoUnit.getLecture().getCourse(), null)) {
+            throw new AccessForbiddenException(NOT_ALLOWED);
+        }
+        return ResponseEntity.ok().body(videoUnit);
+    }
+
+    /**
+     * PUT lectures/:lectureId/video-units : Updates an existing video unit .
+     *
+     * @param lectureId      the id of the lecture to which the video unit belongs to update
+     * @param videoUnit the video unit to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated videoUnit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PutMapping("lectures/{lectureId}/video-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<VideoUnit> updateVideoUnit(@PathVariable Long lectureId, @RequestBody VideoUnit videoUnit) throws URISyntaxException {
+        log.debug("REST request to update an video unit : {}", videoUnit);
+        if (videoUnit.getId() == null) {
+            throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "idnull");
+        }
+
+        if (videoUnit.getLecture() == null || videoUnit.getLecture().getCourse() == null) {
+            throw new ConflictException();
+        }
+
+        // Validate the URL
+        try {
+            new URL(videoUnit.getSource());
+        }
+        catch (MalformedURLException exception) {
+            throw new BadRequestAlertException("The URL is not valid", ENTITY_NAME, "invalidurl");
+        }
+
+        if (!authorizationCheckService.isAtLeastEditorInCourse(videoUnit.getLecture().getCourse(), null)) {
+            throw new AccessForbiddenException(NOT_ALLOWED);
+        }
+
+        if (!videoUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException();
+        }
+
+        VideoUnit result = videoUnitRepository.save(videoUnit);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, videoUnit.getId().toString())).body(result);
```

Comment:
I assume the issue originates here:
As the `applicationName` is now `lectureApp`, the translation key `artemisApp.videoUnit.updated` in the green alert that pops up is no longer translated correctly in the client as the key `lectureApp.videoUnit.updated` cannot be found.

Reply:
I will address this issue in [#4466](https://github.com/ls1intum/Artemis/pull/4466). I will update the translations in the client application to use lectureApp, instead of artemisApp.

Reply:
Addressed in [#4466](https://github.com/ls1intum/Artemis/pull/4466/commits/568ef4cc8b60e5ab1b64819faf5829cfecea32dd)